src/transducer.js

```diff
@@ -1,40 +1,93 @@
-let Immutable = require('immutable');
-
-let data = require('../data/example_data.json');
-
-let things_we_have = Immutable.Set(["pebble time", "nightscout"]);
-let everything = []
-let none = []
-let some = []
-
-//data is pulled from example_data.json which isincluded in index.html
-for (let project of data.projects) {
-  console.log(project);
-  for (let dependencies of project.configurations) {
-	  let dep = Immutable.Set(dependencies);
-
-	  let intersection = dep.intersect(things_we_have)
-	  //intersection of have,dep = dep => everything
-	  if (intersection.equals(dep)){
-		  console.log("Everything")
-      everything.push(project.name)
-	  }
-	  //intersection of have,dep is empty => none
-	  else if(intersection.isEmpty()) {
-		  console.log("None")
-      none.push(project.name)
-	  }
-	  //intersection of have,dep is subset of r => some
-	  else if(intersection.isSubset(dep)){
-		  console.log("Some")
-      some.push(project.name)
-	  }
+import Immutable from 'immutable';
+
+import * as data from '../data/data.json';
+
+/*
+ * Stupid name for a thing that takes a list of selected requirements and
+ * determines which projects can be built with the given list.
+ */
+class Transducer {
+  constructor() {
+    this.projects = Transducer.flattenProjects();
+  }
+
+  /*
+   * In `data.json`, projects are nested by category. This flattens the list.
+   * @returns {Immutable.Set}
+   */
+  static flattenProjects() {
+    const projects = Immutable.Map(data.projects);
+    return projects.reduce(((acc, projects, category) =>
+      acc.union(Immutable.Set(projects))),
+      Immutable.Set());
+  }
+
+  /*
+   * Determine which projects for which we have all, some, and no dependencies.
+   * @param {Immutable.Set} userDeps - the set of selected requirements.
+   * @returns {[Immutable.Set, Immutable.Set, Immutable.Set]}
+   */
+  considerDependencies(userDeps) {
```

Comment:
A better approach would be using [Immutable#groupBy](https://facebook.github.io/immutable-js/docs/#/Set/groupBy) to partition each configuration's dependency set and returning an object like `{all: ..., some: ..., none: ...}` from this function. This also solves the problem of passing `allDeps` and friends all the way from the `ProjectSection` to the `Project` component.